gax/src/test/java/com/google/api/gax/rpc/mtls/MtlsUtilsTest.java

```diff
@@ -0,0 +1,202 @@
+      return value;
+    }
+  }
+
+  @Test
+  public void testUseMtlsClientCertificateEmpty() {
+    MtlsProvider mtlsProvider =
+        new MtlsUtils.DefaultMtlsProvider(new TestEnvironmentProvider(""), "/path/to/missing/file");
+    assertFalse(mtlsProvider.useMtlsClientCertificate());
+  }
+
+  @Test
+  public void testUseMtlsClientCertificateNull() {
+    MtlsProvider mtlsProvider =
+        new MtlsUtils.DefaultMtlsProvider(
+            new TestEnvironmentProvider(null), "/path/to/missing/file");
+    assertFalse(mtlsProvider.useMtlsClientCertificate());
+  }
+
+  @Test
+  public void testUseMtlsClientCertificateTrue() {
+    MtlsProvider mtlsProvider =
+        new MtlsUtils.DefaultMtlsProvider(
+            new TestEnvironmentProvider("true"), "/path/to/missing/file");
+    assertTrue(mtlsProvider.useMtlsClientCertificate());
+  }
+
+  @Test
+  public void testLoadDefaultKeyStoreMissingFile()
+      throws InterruptedException, GeneralSecurityException, IOException {
+    MtlsProvider mtlsProvider =
+        new MtlsUtils.DefaultMtlsProvider(
+            new TestEnvironmentProvider("true"), "/path/to/missing/file");
+    KeyStore keyStore = mtlsProvider.getKeyStore();
+    assertNull(keyStore);
+  }
+
+  @Test
+  public void testLoadDefaultKeyStore()
+      throws InterruptedException, GeneralSecurityException, IOException {
+    MtlsProvider mtlsProvider =
+        new MtlsUtils.DefaultMtlsProvider(
+            new TestEnvironmentProvider("true"),
+            "src/test/resources/com/google/api/gax/rpc/mtls/mtls_context_aware_metadata.json");
+    KeyStore keyStore = mtlsProvider.getKeyStore();
+    assertNotNull(keyStore);
+  }
+
+  @Test
+  public void testLoadDefaultKeyStoreBadCertificate()
+      throws InterruptedException, GeneralSecurityException, IOException {
+    MtlsProvider mtlsProvider =
+        new MtlsUtils.DefaultMtlsProvider(
+            new TestEnvironmentProvider("true"),
+            "src/test/resources/com/google/api/gax/rpc/mtls/mtls_context_aware_metadata_bad_command.json");
+    try {
+      mtlsProvider.getKeyStore();
+      fail("should throw and exception");
+    } catch (IllegalArgumentException e) {
+      assertTrue(
+          "expected to fail with certificate is missing",
+          e.getMessage().contains("certificate is missing"));
+    }
+  }
+
+  @Test
+  public void testExtractCertificateProviderCommand() throws IOException {
+    InputStream inputStream =
+        this.getClass()
+            .getClassLoader()
+            .getResourceAsStream("com/google/api/gax/rpc/mtls/mtls_context_aware_metadata.json");
+    List<String> command =
+        MtlsUtils.DefaultMtlsProvider.extractCertificateProviderCommand(inputStream);
+    assertEquals(2, command.size());
+    assertEquals("cat", command.get(0));
+    assertEquals(
+        "src/test/resources/com/google/api/gax/rpc/mtls/mtlsCertAndKey.pem", command.get(1));
+  }
+
+  static class TestCertProviderCommandProcess extends Process {
```

Comment:
Minor: please move the helper internal class declaration to either beginning or the end of the test class.

Reply:
done